Composer/packages/lib/indexers/src/utils/lgUtil.ts

```diff
@@ -32,59 +32,73 @@ export function increaseNameUtilNotExist(templates: LgTemplate[], name: string):
 export function updateTemplate(
   content: string,
   templateName: string,
-  { name, parameters, body }: { name?: string; parameters?: string[]; body?: string }
-): string {
-  const resource = Templates.parseText(content);
+  { name, parameters, body }: { name?: string; parameters?: string[]; body?: string },
+  importResolver?: ImportResolverDelegate
+): Templates {
+  const resource = Templates.parseText(content, undefined, importResolver);
   const originTemplate = resource.toArray().find((t) => t.name === templateName);
   // add if not exist
   if (!originTemplate) {
-    return resource.addTemplate(templateName, parameters || [], body || '').toString();
+    return resource.addTemplate(templateName, parameters || [], body || '');
   } else {
-    return resource
-      .updateTemplate(
-        templateName,
-        name || originTemplate.name,
-        parameters || originTemplate.parameters,
-        body || originTemplate.body
-      )
-      .toString();
+    return resource.updateTemplate(
+      templateName,
+      name || originTemplate.name,
+      parameters || originTemplate.parameters,
+      body || originTemplate.body
+    );
   }
 }
 
 // if name exist, throw error.
-export function addTemplate(content: string, { name, parameters = [], body }: LgTemplate): string {
-  const resource = Templates.parseText(content);
-  return resource.addTemplate(name, parameters, body).toString();
+export function addTemplate(
+  content: string,
+  { name, parameters = [], body }: LgTemplate,
+  importResolver?: ImportResolverDelegate
+): Templates {
+  const resource = Templates.parseText(content, undefined, importResolver);
+  return resource.addTemplate(name, parameters, body);
 }
 
 // if name exist, add it anyway, with name like `${name}1` `${name}2`
 export function addTemplateAnyway(
   content: string,
-  { name = 'TemplateName', parameters = [], body = '-TemplateBody' }: LgTemplate
-): string {
-  const resource = Templates.parseText(content);
+  { name = 'TemplateName', parameters = [], body = '-TemplateBody' }: LgTemplate,
+  importResolver?: ImportResolverDelegate
+): Templates {
+  const resource = Templates.parseText(content, undefined, importResolver);
   const newName = increaseNameUtilNotExist(resource.toArray(), name);
 
-  return resource.addTemplate(newName, parameters, body).toString();
```

Comment:
Seems all lgUtil methods return value has changed from `string` to `Templates`. That's great! since we don't need pass them cross worker & iframe, which those objects are non-copy-able.

If we step forward, can we return `LgFile` type instead of `Templates` type from lgParser, cause I have saw in lgIndexer there added a method `convertTemplatesToLgFile`, I image all scenario call lgUtil would also call parse to  get a `LgFile`.